File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/FlowTestCase.java
package com.raizlabs.android.dbflow.test;

import android.content.Context;
import android.os.Build;

import org.junit.Rule;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

/**
 * Description:
 */
@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = Build.VERSION_CODES.LOLLIPOP)
public abstract class FlowTestCase {

    @Rule
    public DBFlowTestRule dbFlowTestRule = DBFlowTestRule.create();

    protected Context getContext() {
        return RuntimeEnvironment.application;
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/FlowTestCase.kt
package com.raizlabs.android.dbflow

import android.content.Context
import android.os.Build

import org.junit.Rule
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

/**
 * Description:
 */
@RunWith(RobolectricTestRunner::class)
@Config(constants = BuildConfig::class, sdk = intArrayOf(Build.VERSION_CODES.LOLLIPOP),
    assetDir = "build/intermediates/classes/test/")
abstract class FlowTestCase {

    @JvmField
    @Rule
    var dbFlowTestRule = DBFlowTestRule.create()

    protected val context: Context
        get() = RuntimeEnvironment.application
}
